lib/log.js

```diff
@@ -1,31 +1,101 @@
 "use strict";
 
-var chalk = require("chalk");
+// Imports note: this file is the _only_ one required by `bin/builder.js` which
+// can be a global or local import. To avoid using old versions of libs before
+// we switch to this local scope, we lazy-require things.
 
 // Wrap "type".
 var wrapType = function (type) {
   return "[builder" + (type ? ":" + type : "") + "]";
 };
 
+// Levels
+var LEVELS = {
+  info: 0,
+  warn: 1,
+  error: 2,
+  none: 3
+};
+
 /**
  * A super-small logger.
  */
-module.exports = {
-  // TODO(6): Configurable log levels.
-  // https://github.com/FormidableLabs/builder/issues/6
-  _logger: function () {
-    return console;
+var log = module.exports = {
+  /**
+   * Set log level from command flags or environment.
+   *
+   * **NOTE - Side Effects**: Mutates environment.
+   *
+   * @param {Object} opts         Options object
+   * @param {Object} opts.argv    Arguments array.
+   * @param {Object} opts.env     Environment object to mutate (Default `process.env`)
+   * @returns {void}
+   */
+  setLevel: function (opts) {
+    opts = opts || {};
+    var env = opts.env && opts.env.env || process.env;
+
+    // Try to determine log level from environment.
+    var level = env._BUILDER_ARGS_LOG_LEVEL;
+
+    // If not, lazy require args and determine log level from command line.
+    if (!level) {
+      var args = require("./args"); // eslint-disable-line global-require
+      var parsed = args.general(opts.argv);
+      level = parsed.quiet === true ? "none" : parsed.logLevel;
+    }
+
+    // Statefully set level.
+    env._BUILDER_ARGS_LOG_LEVEL = level;
+    log._level = LEVELS[level];
+    if (typeof log._level === "undefined") {
+      throw new Error("Unknown log level: " + level);
+    }
+
+    // Drain message queue.
+    if (log._queue) {
```

Comment:
```js
var drainQueue = function drainQueue() {
  ...
}
```

I just have a soft spot for decomposing where possible. If I see a comment above a block of code that effectively "names" it I always think it would be better to extract into a separate function.